### ISSUE TITLE
completing-read with obarray

### ISSUE THREAD

**Issue by haji-ali** [minad/vertico#134]
Running `(completing-read "Test" obarray nil t nil nil "t")` on  my Emacs 27.2, the first item in the completion list  is an empty string instead of "t" which is the second item. I am not 100% sure where the empty item is coming from.

Is this expected?

**Comment by minad** [comment]
> Is this expected?

Yes. You can create an empty symbol with `(intern "")`. Using `obarray` without a predicate is not particularly meaningful.

**Comment by minad** [comment]
There was still some odd behavior if empty candidates are present.

~~~ elisp
(completing-read "Test: " '("" "x") nil t nil nil "t")
(completing-read "Test: " '("" "x" "t") nil t nil nil "t")
(completing-read "Test: " '("" "x") nil t)
~~~

**Comment by haji-ali** [comment]
Thanks @minad. I noticed this problem when using `helpful`'s describe variable where the predicate `helpful--variable-p`, returns t for `(intern "")`. For some reasons, I have this variable defined and it ends up on the top of my list of variables regardless of what the default variable I pass.

I am not sure which oddity is resolved by 25f1ed19676829d146141afd0ad42ee4efdb3, but even with this fix, executing 

```
(completing-read "Test: " '("" "x" "t") nil t nil nil "t")
```
still selects the empty string by default rather than "t" (but if I press RET immediately, "t" is correctly returned).

**Comment by minad** [comment]
Yes, it appears at the top since "" is a valid return value and the current input takes precedence over the default value if it matches the input exactly. However even if you select "", in the end the default value is returned since the empty string is always treated as the default. So in other words - empty strings can be considered invalid candidates. I suggest you get rid of your empty variable then or patch `helpful-predicate-p` ;)
